src/assets.js

```diff
@@ -730,6 +769,66 @@ class Assets extends Events {
 
         assets.forEach(a => this.remove(a));
     }
+
+    /**
+     * Gets the default callback called when on asset upload succeeds.
+     * The function takes 2 arguments: the upload id, and the new asset.
+     *
+     * @type {Function<number, Asset>}
+     */
+    get defaultUploadCompletedCallback() {
+        return this._defaultUploadCompletedCallback;
+    }
+
+    /**
+     * Sets the default callback called when on asset upload succeeds.
+     * The function takes 2 arguments: the upload id, and the new asset.
+     *
+     * @type {Function<number, Asset>}
+     */
+    set defaultUploadCompletedCallback(value) {
+        this._defaultUploadCompletedCallback = value;
```

Comment:
What are these used for as by default they are null?

Reply:
They are used in the editor to report progress so that if someone uses the functions in the console they won't have to do anything for the editor to show progress bars etc

Reply:
What happens if there are multiple uploads done separately? eg I start a second upload while the first one is still uploading. Are the progress values of each upload averaged out or combined in some way?

Reply:
Yeah it's handled by the Editor.